test/fixture/locators.test.ts

```diff
@@ -0,0 +1,138 @@
+
+    const element = getByText('Hello h1')
+
+    expect(await element.textContent()).toEqual('Hello h1')
+  })
+
+  test('should handle the get* method failures', async ({queries}) => {
+    const {getByTitle} = queries
+    // Use the scoped element so the pretty HTML snapshot is smaller
+
+    await expect(async () => getByTitle('missing').textContent()).rejects.toThrow()
+  })
+
+  test('should handle the LabelText methods', async ({queries}) => {
+    const {getByLabelText} = queries
+    const element = getByLabelText('Label A')
+
+    /* istanbul ignore next */
+    expect(await element.evaluate(el => el.outerHTML)).toMatch(
+      `<input id="label-text-input" type="text">`,
+    )
+  })
+
+  test('should handle the queryAll* methods', async ({queries}) => {
+    const {queryAllByText} = queries
+    const elements = queryAllByText(/Hello/)
+
+    expect(await elements.count()).toEqual(3)
+
+    const text = await Promise.all([
+      elements.nth(0).textContent(),
+      elements.nth(1).textContent(),
+      elements.nth(2).textContent(),
+    ])
+
+    expect(text).toEqual(['Hello h1', 'Hello h2', 'Hello h3'])
+  })
+
+  test('should handle the queryAll* methods with a selector', async ({queries}) => {
+    const {queryAllByText} = queries
+    const elements = queryAllByText(/Hello/, {selector: 'h2'})
+
+    expect(await elements.count()).toEqual(1)
+
+    expect(await elements.textContent()).toEqual('Hello h2')
+  })
+
+  test('should handle the getBy* methods with a selector', async ({queries}) => {
+    const {getByText} = queries
+    const element = getByText(/Hello/, {selector: 'h2'})
+
+    expect(await element.textContent()).toEqual('Hello h2')
+  })
+
+  test('should handle the getBy* methods with a regex name', async ({queries}) => {
+    const {getByRole} = queries
+    const element = getByRole('button', {name: /getBy.*Test/})
+
+    expect(await element.textContent()).toEqual('getByRole Test')
+  })
+
+  test('supports `hidden` option when querying by role', async ({queries: {queryAllByRole}}) => {
+    const elements = queryAllByRole('img')
+    const hiddenElements = queryAllByRole('img', {hidden: true})
+
+    expect(await elements.count()).toEqual(1)
+    expect(await hiddenElements.count()).toEqual(2)
+  })
+
+  test.describe('querying by role with `level` option', () => {
+    test('retrieves the correct elements when querying all by role', async ({
+      queries: {queryAllByRole},
+    }) => {
+      const elements = queryAllByRole('heading')
+      const levelOneElements = queryAllByRole('heading', {level: 3})
+
+      expect(await elements.count()).toEqual(3)
+      expect(await levelOneElements.count()).toEqual(1)
+    })
+
+    test('does not throw when querying for a specific element', async ({queries: {getByRole}}) => {
+      expect.assertions(1)
+
+      await expect(getByRole('heading', {level: 3}).textContent()).resolves.not.toThrow()
+    })
+  })
+
+  // TODO: scoped queries with `within`
```

Comment:
How are you thinking this API will work? Will it take `Locator`s and return the queries interface?

```ts
const messagesLocator = screen.getByText('messages')
const helloMessageLocator = within(messagesLocator).getByText('hello')

expect(await helloMessageLocator.count()).toEqual(1)
```

Reply:
The [documentation](https://playwright.dev/docs/extensibility#custom-selector-engines) mentions a `create` method, which I was thinking we might be able to use to scope queries but it doesn't show an implementation in the example. I'm hoping we can just leverage the combinator here.